packages/lib-classifier/src/store/SubjectStore.spec.js

```diff
@@ -51,97 +46,78 @@ describe('Model > SubjectStore', function () {
 
     describe('with less than three subjects in the queue', function () {
       let populateSpy
+      let rootStore
 
       before(function () {
+        rootStore = setupStores()
         populateSpy = sinon.spy(rootStore.subjects, 'populateQueue')
```

Comment:
Here, the queue should be populated, then advanced until there are less than 3 subjects.

Previously, a populated queue was set up in the outer function, then advanced to the required state by these inner functions. Now, each inner function is setting up a new queue to test so the initial conditions for each test have to be set up too. Otherwise, you're just testing the initial state of an empty queue.

Reply:
The while loop doesn't run in the before block because it gets called before the async populate queue call is done in the setup. It does run before the expectation is called in the it statement. I could try putting it in a `beforeEach` instead.